### PR TITLE
 Enhanced docstring

### PR DESCRIPTION
**Pull request by abhi-glitchhg**
During the last PR (#45), I had to revert back because of compatibility issues

In this PR I have added some docstrings and Minor changes like changing variable names

this PR is the same as - #48 with edited title :) 

@NeelayS

### CODE REVIEW

File: vformer/attention/spatial.py [hunk 1]
```diff
@@ -69,6 +69,20 @@ def __init__(
             self.sr = nn.Conv2d(dim, dim, kernel_size=1, stride=1)
 
     def forward(self, x, H, W):
+        """
+
+        Args:
```
Comment (NeelayS) [inline vformer/attention/spatial.py:74]: We are following NumPy-style, so this needs to be the following way, doesn't it?

```suggestion
        Parameters
        --------------
```

File: vformer/attention/spatial.py [hunk 2]
```diff
@@ -69,6 +69,20 @@ def __init__(
...
+            x: torch.Tensor
+                Input tensor
+            H: int
+                Height  of image patches
+            W: int
+                Width of image patches
+
+        Returns: torch.Tensor
```
Comment (NeelayS) [inline vformer/attention/spatial.py:82]:
```suggestion
        Returns
        ----------
        
        torch.Tensor
```

**Review by NeelayS — CHANGES_REQUESTED** [review]
Left a comment.

**Comment by NeelayS** [comment]
Also, you can edit the name of a PR after opening it by clicking the edit button besides it.

**Comment by abhi-glitchhg** [comment]
> 

F, sorry :/

File: vformer/attention/spatial.py [hunk 3]
```diff
@@ -69,6 +69,23 @@ def __init__(
...
+                Input tensor
+            H: int
+                Height  of image patches
+            W: int
+                Width of image patches
+
+        Returns
+        ----------
+            torch.Tensor
+            Returns output tensor by applying spatial attention on input tensor
```
Comment (NeelayS) [inline vformer/attention/spatial.py:86]:
```suggestion
                Returns output tensor by applying spatial attention on input tensor
```

File: vformer/attention/spatial.py [hunk 4]
```diff
@@ -69,6 +69,23 @@ def __init__(
             self.sr = nn.Conv2d(dim, dim, kernel_size=1, stride=1)
 
     def forward(self, x, H, W):
+        """
+
+        Parameters
+        ----------
+            x: torch.Tensor
```
Comment (NeelayS) [inline vformer/attention/spatial.py:76]: The parameter names listed can start in the same column as the Parameters heading. They don't need to be indented.

Comment (NeelayS) [inline vformer/attention/spatial.py:76]:
```suggestion
        x: torch.Tensor
```

File: vformer/attention/window.py [hunk 5]
```diff
@@ -57,6 +59,23 @@ def __init__(
...
+        """
+
+        Parameters
+        ----------
+            x: torch.Tensor
+                input Tensor
+            mask: torch.Tensor
+                Attention mask used for shifted window attention, if None, window attention will be used,
+                else attention mask will be taken into consideration.
+                for more understanding you may refer (this)[https://github.com/microsoft/Swin-Transformer/issues/38]
```
Comment (NeelayS) [inline vformer/attention/window.py:71]: For this link to render properly in the documentation, you will have to use the RST format of specifying links.

File: vformer/encoder/pyramid.py [hunk 6]
```diff
@@ -65,6 +66,25 @@ def __init__(
         )
 
     def forward(self, x, **kwargs):
+        """
+
+        Parameters
+        ----------
+            x: torch.Tensor
+                Input tensor
+        Kwargs
```
Comment (NeelayS) [inline vformer/encoder/pyramid.py:75]: You don't need to have a separate section for **kwargs. Check [this](https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#:~:text=length%20argument%20list.-,**kwargs,-Arbitrary%20keyword%20arguments)

**Review by NeelayS — CHANGES_REQUESTED** [review]
Have added a few suggestions. Please check.

**Comment by NeelayS** [comment]
@abhi-glitchhg

**Pull request closed without merging**